### PR TITLE
Rebase nuttx

### PR DESCRIPTION
**Pull request by jlaitine**
Rebase NuttX again to latest upstream master.

Temporarily disable the Athena HW crypto block from v2 and pi bootloaders, as nuttx size has grown so much that it doesn't fit otherwise. This needs to be resolved

### CODE REVIEW

**Review by pussuw — APPROVED** [review]

File: src/include/visibility.h [hunk 1]
```diff
@@ -60,7 +60,7 @@
 #endif
 
 
-#define system_exit exit
```
Comment (pussuw) [inline src/include/visibility.h:63]: Is this called from user and kernel ?

File: platforms/nuttx/src/px4/common/tasks.cpp [hunk 2]
```diff
@@ -92,7 +92,7 @@ int px4_task_delete(int pid)
 
 	if (pid == getpid()) {
 		// Commit suicide
-		exit(EXIT_SUCCESS);
```
Comment (pussuw) [inline platforms/nuttx/src/px4/common/tasks.cpp:95]: This needs to be flagged with __KERNEL__. I think user space should still call exit() ?

**Review by pussuw — APPROVED** [review]
Blah, looks good to me. The code for px4_task_delete changes with kernel mode, and the define macro needs to be verified it is used correctly. I can do that.

**Comment by jlaitine** [comment]
OK, merging, thanks!

**Pull request merged**